allauth/account/forms.py

```diff
@@ -327,12 +327,10 @@ def custom_signup(self, request, user):
 
 
 class SignupForm(BaseSignupForm):
-
-    password1 = PasswordField(label=_("Password"))
-    password2 = PasswordField(label=_("Password (again)"))
-
     def __init__(self, *args, **kwargs):
         super(SignupForm, self).__init__(*args, **kwargs)
+        self.fields['password1'] = PasswordField(label=_("Password"))
+        self.fields['password2'] = PasswordField(label=_("Password (again)"))
         if not app_settings.SIGNUP_PASSWORD_ENTER_TWICE:
             del self.fields["password2"]
```

Comment:
It makes little sense to first add the field only to remove it... I would turn this `if not` into an `if` and add the field in there.